dataverse-webapp/src/main/java/edu/harvard/iq/dataverse/notification/UserNotificationService.java

```diff
@@ -0,0 +1,145 @@
+import edu.harvard.iq.dataverse.persistence.user.AuthenticatedUser;
+import edu.harvard.iq.dataverse.persistence.user.NotificationType;
+import edu.harvard.iq.dataverse.persistence.user.UserNotification;
+import edu.harvard.iq.dataverse.persistence.user.UserNotificationDao;
+
+import javax.ejb.Stateless;
+import javax.inject.Inject;
+import java.sql.Timestamp;
+import java.util.concurrent.ExecutorService;
+import java.util.concurrent.Executors;
+
+/**
+ * Class responsible for managing user notifications.
+ */
+@Stateless
+public class UserNotificationService {
+
+    private UserNotificationDao userNotificationDao;
+    private MailService mailService;
+    private EmailNotificationMapper mailMapper;
+
+    private ExecutorService executorService;
+
+    // -------------------- CONSTRUCTORS --------------------
+    @Deprecated /* JEE requirement*/
+    UserNotificationService() {
+    }
+
+    @Inject
+    public UserNotificationService(UserNotificationDao userNotificationDao, MailService mailService, EmailNotificationMapper mailMapper) {
+        this.userNotificationDao = userNotificationDao;
+        this.mailService = mailService;
+        this.mailMapper = mailMapper;
+
+        executorService = Executors.newFixedThreadPool(Runtime.getRuntime().availableProcessors());
+    }
+
+    // -------------------- LOGIC --------------------
+
+    /**
+     * Saves notification to database, hereby sends notification to users dashboard.
+     */
+    public void sendNotification(AuthenticatedUser dataverseUser, Timestamp sendDate, NotificationType type) {
+        UserNotification userNotification = createUserNotification(dataverseUser, sendDate, type);
+
+        userNotificationDao.save(userNotification);
+    }
+
+    /**
+     * Saves notification to database, then sends email asynchronously.
+     *
+     * @param notificationObjectType - type has to match correct #{@link NotificationType}
+     */
+    public void sendNotificationWithEmail(AuthenticatedUser dataverseUser,
+                                          Timestamp sendDate,
+                                          NotificationType type,
+                                          long dvObjectId,
+                                          NotificationObjectType notificationObjectType) {
+
+        UserNotification userNotification = createUserNotification(dataverseUser, sendDate, type, dvObjectId);
+
+        userNotificationDao.save(userNotification);
+        userNotificationDao.flush();
+
+        EmailNotificationDto emailNotificationDto = mailMapper.toDto(userNotification, dvObjectId, notificationObjectType);
+
+        executorService.submit(() -> sendEmail(emailNotificationDto));
+    }
+
+    /**
+     * Saves notification to database, then sends email asynchronously.
+     * @param notificationObjectType - type has to match correct #{@link NotificationType}
+     */
+    public void sendNotificationWithEmail(AuthenticatedUser dataverseUser,
```

Comment:
Should this method body be like: 
`sendNotification(...);`
`sendEmail(...);` 
?